lang/LangPrimSource/PyrListPrim.cpp

```diff
@@ -492,41 +494,123 @@ int prEvent_Delta(struct VMGlobals *g, int numArgsPushed)
 	PyrSlot *a, key, dur, stretch, delta;
 	double fdur, fstretch;
 	int err;
+	PyrClass *restClass = getsym("Rest")->u.classobj;
+	PyrSlot *slot;
 
 	a = g->sp;  // dict
 
 	SetSymbol(&key, s_delta);
 	identDict_lookup(slotRawObject(a), &key, calcHash(&key), &delta);
 
 	if (NotNil(&delta)) {
-		slotCopy(a,&delta);
+		if (isKindOfSlot(&delta, restClass)) {
+			slot = slotRawObject(&delta)->slots;
+			err = slotDoubleVal(slot, &fdur);
+		} else {
+			err = slotDoubleVal(&delta, &fdur);
+		};
+		if (err) {
+			return err;
+		} else {
+			SetFloat(g->sp, fdur);
+			return errNone;
+		}
 	} else {
 		SetSymbol(&key, s_dur);
 		identDict_lookup(slotRawObject(a), &key, calcHash(&key), &dur);
-
 		err = slotDoubleVal(&dur, &fdur);
 		if (err) {
-			if (NotNil(&dur)) return err;
-			SetNil(a);
-			return errNone;
+			if (NotNil(&dur)) {
+				if (isKindOfSlot(&dur, restClass)) {
+					slot = slotRawObject(&dur)->slots;
+					err = slotDoubleVal(slot, &fdur);
+					if (err) return err;
+				} else {
+					return errWrongType;
+				}
+			} else {
+				SetNil(g->sp);
+				return errNone;
+			}
 		}
 
 		SetSymbol(&key, s_stretch);
 		identDict_lookup(slotRawObject(a), &key, calcHash(&key), &stretch);
 
 		err = slotDoubleVal(&stretch, &fstretch);
 		if (err) {
-			if (NotNil(&stretch)) return err;
-			SetFloat(a, fdur);
-			return errNone;
+			if (NotNil(&stretch)) {
+				if (isKindOfSlot(&stretch, restClass)) {
+					slot = slotRawObject(&stretch)->slots;
+					err = slotDoubleVal(slot, &fstretch);
+					if (err) return err;
+				} else {
+					return errWrongType;
+				}
+			} else {
+				SetFloat(g->sp, fdur);
+				return errNone;
+			}
 		}
 
-		SetFloat(a, fdur * fstretch );
+		SetFloat(g->sp, fdur * fstretch);
 	}
 
 	return errNone;
 }
 
+int prEvent_IsRest(struct VMGlobals *g, int numArgsPushed);
+int prEvent_IsRest(struct VMGlobals *g, int numArgsPushed)
+{
+	PyrSlot *dictslots = slotRawObject(g->sp)->slots;
+	PyrSlot *arraySlot = dictslots + ivxIdentDict_array;
+
+	if (isKindOfSlot(arraySlot, class_array)) {
+		PyrSlot key, typeSlot;
+		PyrSymbol *typeSym;
+		// test 'this[\type] == \rest' first
+		SetSymbol(&key, s_type);
+		identDict_lookup(slotRawObject(g->sp), &key, calcHash(&key), &typeSlot);
+		if(!slotSymbolVal(&typeSlot, &typeSym) && typeSym == s_rest) {
+			SetBool(g->sp, 1);
+			return errNone;
+		} else {
+			PyrObject *array = slotRawObject(arraySlot);
+			PyrSymbol *slotSym;
+			PyrSlot *slot;
+			int32 size = array->size;
+			int32 i;
+
+			slot = array->slots + 1;  // scan only the odd items
+
+			for (i = 1; i < size; i += 2, slot += 2) {
+				if (isKindOfSlot(slot, class_rest)
+				    || isKindOfSlot(slot, class_metarest)
```

Comment:
I could see two options to warn users about using the Rest class directly:

1. Check yielded values in the language. I think that would be going back to `processRest` territory, which we wanted to avoid.

2. Print the warning from within the primitive... unusual for SC but it would have the smallest code footprint.

I guess I would lean toward the second.